rust/src/data/ffi.rs

```diff
@@ -139,9 +145,46 @@ pub extern "C" fn opendp_data__slice_as_object(
             .collect::<HashMap<K, V>>();
         Ok(AnyObject::new(map))
     }
+
+    #[cfg(feature = "polars")]
+    pub fn raw_to_concrete_series(
+        raw: &FfiSlice
+    ) -> Fallible<Series> {
+        let slice = unsafe { slice::from_raw_parts(raw.ptr as *const *const c_void, raw.len) };
+        if slice.len() != 3 {
+            return fallible!(FFI, "Series FfiSlice must have length 3");
+        }
+        Ok(unsafe {
+            // consume the arrow array eagerly
+            let array = *Box::from_raw(slice[0] as *mut ArrowArray);
+            let schema = try_as_ref!(slice[1] as *const ArrowSchema);
+            let name = util::to_str(slice[2] as *const c_char)?;
+
+            let field = arrow::ffi::import_field_from_c(schema)
+                .map_err(|e| err!(FFI, "failed to import field from c: {}", e.to_string()))?;
+            let array = arrow::ffi::import_array_from_c(array, field.data_type)
+                .map_err(|e| err!(FFI, "failed to import array from c: {}", e.to_string()))?;
+            Series::try_from((name, array))
+                .map_err(|e| err!(FFI, "failed to construct Series: {}", e.to_string()))?
+        })
+    }
+    #[cfg(feature = "polars")]
+    pub fn raw_to_series(
+        raw: &FfiSlice
+    ) -> Fallible<AnyObject> {
+        raw_to_concrete_series(raw).map(AnyObject::new)
+    }
+
+
     match T.contents {
         TypeContents::PLAIN("String") => raw_to_string(raw),
         TypeContents::PLAIN("ExtrinsicObject") => raw_to_plain::<ExtrinsicObject>(raw),
+
+        #[cfg(feature = "polars")]
+        TypeContents::PLAIN("Series") => raw_to_series(raw),
+        #[cfg(feature = "polars")]
+        TypeContents::PLAIN("SeriesDomain") => raw_to_series(raw),
```

Comment:
```suggestion
        TypeContents::PLAIN("Series") | TypeContents::PLAIN("SeriesDomain") => raw_to_series(raw),
```
Would that do the same thing?

Seeing this repeated on separate lines, I'm wondering if there is a typo and you meant the RHS to be different.

Reply:
Good catch. Removed as it is not necessary.